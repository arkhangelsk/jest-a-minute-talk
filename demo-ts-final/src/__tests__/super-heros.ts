import { getFlyingSuperHeros } from "../super-heros";

test('manual "snapshot"', () => {
  const flyingHeros = getFlyingSuperHeros();
  expect(flyingHeros).toEqual([
    { name: "Dynaguy", powers: ["disintegration ray", "fly"] },
    { name: "Apogee", powers: ["gravity control", "fly"] }
  ]);
});

test("automatic snapshot", () => {
  const flyingHeros = getFlyingSuperHeros();
  expect(flyingHeros).toMatchSnapshot();
});

test("inline snapshot", () => {
  const flyingHeros = getFlyingSuperHeros();
  expect(flyingHeros).toMatchInlineSnapshot(`
Array [
  Object {
    "name": "Dynaguy",
    "powers": Array [
      "disintegration ray",
      "fly",
    ],
  },
  Object {
    "name": "Apogee",
    "powers": Array [
      "gravity control",
      "fly",
    ],
  },
]
`);
});
