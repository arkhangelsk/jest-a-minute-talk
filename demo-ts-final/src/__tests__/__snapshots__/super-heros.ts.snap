// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`automatic snapshot 1`] = `
Array [
  Object {
    "name": "Dynaguy",
    "powers": Array [
      "disintegration ray",
      "fly",
    ],
  },
  Object {
    "name": "Apogee",
    "powers": Array [
      "gravity control",
      "fly",
    ],
  },
]
`;
